### ISSUE TITLE
Get raw logits per pixel

### ISSUE THREAD

**Issue by NielsRogge** [facebookresearch/MaskFormer#60]
Hi,

Thanks for the great work. I was wondering if it is possible to get raw, unnormalized scores per pixel for the semantic segmentation model. For instance, in [mmseg](https://github.com/open-mmlab/mmsegmentation), all models return `logits` of shape (batch_size, num_labels, height, width). Is this possible with MaskFormer too?

Thank you.

Kind regards,

Niels

**Comment by bowenc0221** [comment]
The output of MaskFormer is already in raw `logits` (unnormalized scores) of shape [num_labels, height, width] (because the detectron2 evaluator takes 1 image at a time): https://github.com/facebookresearch/MaskFormer/blob/da3e60d85fdeedcb31476b5edd7d328826ce56cc/mask_former/mask_former_model.py#L242-L246

The evaluator simply takes `argmax` on the raw `logits` to get labels and we never use normalized scores.

**Comment by bowenc0221** [comment]
I just saw your discussion on the definition of `logits` in https://github.com/huggingface/transformers/pull/15469#discussion_r798362948.

If you are referring `logits` to the value before taking `softmax`, then it may not be possible to get `logits` from MaskFormer.

IMO, `logits` can only be acquired from a per-pixel classification model, but MaskFormer addresses semantic segmentation with a mask classification paradigm instead.

**Comment by NielsRogge** [comment]
Thanks for your answers, very helpful! Expect the model in Huggingface Transformers soon.
